### ISSUE TITLE
Hovers with markdown starting with `- \n` do not appear

### ISSUE THREAD

**Issue by unrelentingtech** [kak-lsp/kak-lsp#329]
https://github.com/haskell/haskell-language-server currently joins parts of the hover info with `"- \n"`, for example the result can be:

```json
{
  "result": {
    "contents": {
      "kind": "markdown",
      "value": "- \nGHC.Int"
    },
    "range": {
      "start": {
        "line": 8,
        "character": 7
      },
      "end": {
        "line": 8,
        "character": 14
      }
    }
  },
  "jsonrpc": "2.0",
  "id": 53
}
```

and kak-lsp 8.0.0 just quietly ignores these hovers. Patching hls to join with spaces avoids the bug.

**Comment by krobelus** [comment]
Can you give an example haskell file where this happens?
Regardless, this will probably be fixed by #513

**Comment by krobelus** [comment]
should be fixed with the new markdown parsing
